### PR TITLE
Created Pull Request Failing and Passing Labels

### PR DESCRIPTION
**Pull request by RachalCassity**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

This pull request introduces a new feature to automatically add labels to pull requests when check statuses fail. The primary goal is to enhance visibility and streamline the process of identifying and addressing failed checks during the review process. Due to the recent changes regarding backend approval, reviewers cannot clearly determine which pull requests have passing checks.

**Label Changes**
Pull requests with a failure label will not be reviewed.

New PR Labels:
- audit-service-failure
- pe-failure
- code-health-failure
- codeowners-addition-failure
- codeowners-delete-failure
- lint-failure
- test-failure
- danger-alert ( Just an alert, pull request can be reviewed)

**Ready for Review**
A github action will run when a PR is added/removed from a PR. The action ensures there are no failure labels assigned. A ready-for-backend-review label will be added to the Pr when there are no failures.

**Require approval changes**

Prevented multiple "This PR requires approval" comments by checking if comment exists with the Require backend-review-group approval: find-required-comment.
When `steps.find-required-comment.outputs.comment-id == ''` the comment will be added to the pull request.
Screenshot 2024-08-06 at 10 26 17 AM

- 
## Testing done

- Tested to ensure labels are properly added and removed

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
* Helps backend developers review passing pull requests

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: .github/workflows/audit_service_tags.yml [hunk 1]
```diff
@@ -77,3 +77,33 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Remove Failure label
+        uses: actions-ecosystem/action-remove-labels@v1
+        with:
+          number: ${{ github.event.pull_request.number }}
+          labels: |
+            audit-service-failure
```
Comment (ericboehs) [inline .github/workflows/audit_service_tags.yml:109]: Missing final new line. https://stackoverflow.com/a/44704969

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -30,3 +30,34 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Remove Failure label
+        uses: actions-ecosystem/action-remove-labels@v1
+        with:
+          number: ${{ github.event.pull_request.number }}
+          labels: |
+            codeql-failure
```
Comment (ericboehs) [inline .github/workflows/codeql-analysis.yml:63]: final new line missing

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -30,3 +30,34 @@ jobs:
         with:
           language: ${{ matrix.language }}
           config-file: ./.github/codeql/config.yml
+
+  assign-failure-label:
+    if: github.event_name == 'pull_request' && failure()
+    needs: [analyze]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```
Comment (ericboehs) [inline .github/workflows/codeql-analysis.yml:39]: Does this need to checkout the repo?

Comment (RachalCassity) [reply .github/workflows/codeql-analysis.yml:39]: Yes

Comment (RachalCassity) [reply .github/workflows/codeql-analysis.yml:39]: Well let me check to see. In the docs, it says to use it but maybe it doesn't.

Comment (RachalCassity) [reply .github/workflows/codeql-analysis.yml:39]: @ericboehs Yes, this is needed

File: .github/workflows/build_atlas_codeowners.yml
Comment (RachalCassity) [inline .github/workflows/build_atlas_codeowners.yml:1]: Removed an additional extra line

File: .github/workflows/ready_for-review.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          labels: |
+            ready-for-backend-review
+
+      - name: Remove Review label
+        uses: actions-ecosystem/action-remove-labels@v1
+        if: env.READY_FOR_REVIEW != false
+        with:
+          number: ${{ github.event.pull_request.number }}
+          labels: |
+            ready-for-backed-review
```
Comment (rmtolmach) [inline .github/workflows/ready_for-review.yml:37]:
```suggestion
            ready-for-backed-review

```

File: .github/workflows/build_atlas_codeowners.yml [hunk 5]
```diff
@@ -48,4 +48,3 @@ jobs:
           user_email: 'devops@va.gov'
           user_name: 'va-vsp-bot'
           commit_message: 'Auto update Atlas with Vets-API CODEOWNERS'
-
```
Comment (RachalCassity) [inline .github/workflows/build_atlas_codeowners.yml:51]: Extra line removed

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**